Training data: 360
Validation data: 240
3D Conv Out: (None, 50, 32, 32, 64)
3D Conv Out Reshape: (None, 32, 32, 64)
Resnet18 Out: (None, 1, 1, 512)
Resnet18 Linear Out: (None, 256)
Input to GRU: (None, 50, 256)
GRU Out: (None, 50, 29)
ResNet LSTM Pretrain weights loaded
outv: (None, 200, 256)
outa: (None, 200, 256)
hidden (None, 256)
attn_out: (None, 200, 512)
attn_states: (None, 200, 200)
fusion: (None, 200, 512)

==================================================================================================
Total params: 29,658,113
Trainable params: 18,093,569
Non-trainable params: 11,564,544
__________________________________________________________________________________________________

Model weights path: test_tdavss_Offset_BothHigh_epochs40_5lr1e-4_exp1

Epoch 1/3
 1/30 [>.............................] - ETA: 3:49 - loss: 48.6101 - snr_acc: -48.6101 2/30 [=>............................] - ETA: 2:22 - loss: 43.4867 - snr_acc: -43.4867 3/30 [==>...........................] - ETA: 1:50 - loss: 39.1763 - snr_acc: -39.1763 4/30 [===>..........................] - ETA: 1:32 - loss: 36.1468 - snr_acc: -36.1468 5/30 [====>.........................] - ETA: 1:22 - loss: 34.1372 - snr_acc: -34.1372 6/30 [=====>........................] - ETA: 1:14 - loss: 33.2554 - snr_acc: -33.2554 7/30 [======>.......................] - ETA: 1:08 - loss: 32.1858 - snr_acc: -32.1858 8/30 [=======>......................] - ETA: 1:03 - loss: 31.1394 - snr_acc: -31.1394 9/30 [========>.....................] - ETA: 58s - loss: 30.3278 - snr_acc: -30.3279 10/30 [=========>....................] - ETA: 54s - loss: 29.3567 - snr_acc: -29.356711/30 [==========>...................] - ETA: 50s - loss: 28.4712 - snr_acc: -28.471212/30 [===========>..................] - ETA: 47s - loss: 27.7150 - snr_acc: -27.715013/30 [============>.................] - ETA: 44s - loss: 26.9646 - snr_acc: -26.964614/30 [=============>................] - ETA: 40s - loss: 26.1854 - snr_acc: -26.185415/30 [==============>...............] - ETA: 38s - loss: 25.5579 - snr_acc: -25.557916/30 [===============>..............] - ETA: 35s - loss: 24.9485 - snr_acc: -24.948517/30 [================>.............] - ETA: 32s - loss: 24.5007 - snr_acc: -24.500718/30 [=================>............] - ETA: 29s - loss: 23.9410 - snr_acc: -23.941019/30 [==================>...........] - ETA: 26s - loss: 23.3388 - snr_acc: -23.338820/30 [===================>..........] - ETA: 24s - loss: 22.8065 - snr_acc: -22.806521/30 [====================>.........] - ETA: 21s - loss: 22.2450 - snr_acc: -22.245022/30 [=====================>........] - ETA: 19s - loss: 21.7083 - snr_acc: -21.708323/30 [======================>.......] - ETA: 16s - loss: 21.1984 - snr_acc: -21.198424/30 [=======================>......] - ETA: 14s - loss: 20.7367 - snr_acc: -20.736725/30 [========================>.....] - ETA: 11s - loss: 20.3722 - snr_acc: -20.372226/30 [=========================>....] - ETA: 9s - loss: 19.9656 - snr_acc: -19.9656 27/30 [==========================>...] - ETA: 7s - loss: 19.6109 - snr_acc: -19.610928/30 [===========================>..] - ETA: 4s - loss: 19.2283 - snr_acc: -19.228329/30 [============================>.] - ETA: 2s - loss: 18.8793 - snr_acc: -18.8793Traceback (most recent call last):
  File "main_tasnet_unsync.py", line 162, in <module>
    callbacks=[reducelronplateau, checkpoint_save_weights, metrics_wandb, LoggingCallback(print_fcn=log_to_file), metrics_unsync], verbose=1)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training.py", line 1297, in fit_generator
    steps_name='steps_per_epoch')
  File "/usr/local/lib/python3.6/dist-packages/wandb/keras/__init__.py", line 104, in new_generator
    return old_generator(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_generator.py", line 332, in model_iteration
    callbacks.on_epoch_end(epoch, epoch_logs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/callbacks.py", line 298, in on_epoch_end
    callback.on_epoch_end(epoch, logs)
  File "/data/AV-speech-separation1/metrics.py", line 472, in on_epoch_end
    preds = self.model.predict(Data_predict_attention(val_folders_samp_dict))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training.py", line 909, in predict
    use_multiprocessing=use_multiprocessing)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2.py", line 462, in predict
    steps=steps, callbacks=callbacks, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2.py", line 444, in _model_iteration
    total_epochs=1)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2.py", line 123, in run_one_epoch
    batch_outs = execution_function(iterator)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2_utils.py", line 86, in execution_function
    distributed_function(input_fn))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/def_function.py", line 457, in __call__
    result = self._call(*args, **kwds)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/def_function.py", line 503, in _call
    self._initialize(args, kwds, add_initializers_to=initializer_map)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/def_function.py", line 408, in _initialize
    *args, **kwds))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/function.py", line 1848, in _get_concrete_function_internal_garbage_collected
    graph_function, _, _ = self._maybe_define_function(args, kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/function.py", line 2150, in _maybe_define_function
    graph_function = self._create_graph_function(args, kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/function.py", line 2041, in _create_graph_function
    capture_by_value=self._capture_by_value),
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/func_graph.py", line 915, in func_graph_from_py_func
    func_outputs = python_func(*func_args, **func_kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/eager/def_function.py", line 358, in wrapped_fn
    return weak_wrapped_fn().__wrapped__(*args, **kwds)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2_utils.py", line 73, in distributed_function
    per_replica_function, args=(model, x, y, sample_weights))
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/distribute/distribute_lib.py", line 760, in experimental_run_v2
    return self._extended.call_for_each_replica(fn, args=args, kwargs=kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/distribute/distribute_lib.py", line 1787, in call_for_each_replica
    return self._call_for_each_replica(fn, args, kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/distribute/distribute_lib.py", line 2132, in _call_for_each_replica
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/autograph/impl/api.py", line 292, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2_utils.py", line 162, in _predict_on_batch
    return predict_on_batch(model, x)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/training_v2_utils.py", line 370, in predict_on_batch
    return model(inputs)  # pylint: disable=not-callable
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/base_layer.py", line 847, in __call__
    outputs = call_fn(cast_inputs, *args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/network.py", line 708, in call
    convert_kwargs_to_constants=base_layer_utils.call_context().saving)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/network.py", line 860, in _run_internal_graph
    output_tensors = layer(computed_tensors, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/base_layer.py", line 847, in __call__
    outputs = call_fn(cast_inputs, *args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/sequential.py", line 270, in call
    outputs = layer(inputs, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/engine/base_layer.py", line 847, in __call__
    outputs = call_fn(cast_inputs, *args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/layers/convolutional.py", line 197, in call
    outputs = self._convolution_op(inputs, self.kernel)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/ops/nn_ops.py", line 1134, in __call__
    return self.conv_op(inp, filter)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/ops/nn_ops.py", line 639, in __call__
    return self.call(inp, filter)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/ops/nn_ops.py", line 238, in __call__
    name=self.name)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/ops/gen_nn_ops.py", line 1553, in conv3d
    dilations=dilations, name=name)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/op_def_library.py", line 631, in _apply_op_helper
    param_name=input_name)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/framework/op_def_library.py", line 60, in _SatisfiesTypeConstraint
    ", ".join(dtypes.as_dtype(x).name for x in allowed_list)))
TypeError: Value passed to parameter 'input' has DataType uint8 not in list of allowed values: float16, bfloat16, float32, float64
